src/PlotDock.cpp

```diff
@@ -825,16 +825,21 @@ void PlotDock::toggleLegendVisible(bool visible)
 void PlotDock::adjustBars()
 {
     const double padding = 0.15;
-    const double groupedWidth = ui->plotWidget->plottableCount()? 1.0 / ui->plotWidget->plottableCount() : 0.0;
+    int plottableCount = ui->plotWidget->plottableCount();
+
+    if (plottableCount == 0)
+        return;
+
+    const double groupedWidth = 1.0 / plottableCount;
     QCPBars* previousBar = nullptr;
-    QCPBarsGroup* barsGroup = m_stackedBars? nullptr : new QCPBarsGroup(ui->plotWidget);
-    for (int i = 0, ie = ui->plotWidget->plottableCount(); i < ie; ++i)
+    QCPBarsGroup* barsGroup = m_stackedBars ? nullptr : new QCPBarsGroup(ui->plotWidget);
+    for (int i = 0, ie = plottableCount; i < ie; ++i)
     {
         QCPBars* bar = qobject_cast<QCPBars*>(ui->plotWidget->plottable(i));
         if (bar) {
             if (m_stackedBars) {
                 // Ungroup if grouped
-                bar->setBarsGroup(nullptr);
+                bar->setBarsGroup(barsGroup);
```

Comment:
What said the analyzer about this line of code? The result will be the same, but I find the previous invocation more explicit, since according to [documentation](https://www.qcustomplot.com/documentation/classQCPBars.html#aedd1709061f0b307c47ddb45e172ef9a):

> To remove this QCPBars from any group, set barsGroup to 0. 

I changed 0 to nullptr only for better type consistency.

Reply:
It actually didn't say anything about this one, but it felt weird to have `QCPBarsGroup* barsGroup = m_stackedBars ? nullptr : new QCPBarsGroup(ui->plotWidget);` where the value (`nullptr`) when `m_stackedBars` is not null is never used.

I think the clearest way would be to replace this with `nullptr` as before and then move `new QCPBarsGroup(ui->plotWidget)` on this line:

https://github.com/sqlitebrowser/sqlitebrowser/blob/b105630475772ac951501e3674d21fd617e4b338/src/PlotDock.cpp#L846

What do you say?

Reply:
But then it would create a new bar group per plottable instead of only one for all. Not sure about the implications of that. I'd have to take a closer look at it later.

Reply:
You're right, sorry! I reverted it.